app/base/static/assets/js/vis/simple-bar-chart.js

```diff
@@ -42,55 +40,112 @@ class SimpleBarChart {
         let svg = d3.select(canvas).append("svg")
             .attr("width", this.CHART_WIDTH)
             .attr("height", this.CHART_HEIGHT - this.GAP);
-
         svg.append("rect")
+            .attr("id", "rect")
             .attr("fill", "#ffffff")
             .attr("width", this.CHART_WIDTH)
             .attr("height", this.CHART_HEIGHT - this.GAP);
 
         let g = svg.append("g")
             .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
 
-        g.append("g")
-            .attr("class", "axis axis--x")
+        //declare xAxis and xAxisE element variable to be used in resize function
+        let xAxis = d3.axisBottom(x)
+            .tickFormat(d3.timeFormat("%Y-%m-%d")).tickValues(x.domain().filter(function (d, i) {
+                return !(i % 10)
+            }));
+        let xAxisEL = g.append("g")
+            .call(xAxis);
+        xAxisEL.attr("class", "axis axis--x")
             .attr("transform", "translate(0," + height + ")")
-            .call(d3.axisBottom(x)
-              .tickFormat(d3.timeFormat("%Y-%m-%d")).tickValues(x.domain().filter(function (d, i) {
-                 return !(i % 10)
-            })))
             .selectAll("text")
             .style("text-anchor", "end")
             .attr("dx", "-.8em")
             .attr("dy", ".15em")
             .attr("transform", "rotate(-65)");
 
-        g.append("g")
-            .attr("class", "axis axis--y")
-            .call(d3.axisLeft(y))
+
+        //declare yAxis and yAxis element variable to be used in resize function
+        let yAxis = d3.axisLeft().scale(y);
+        let yAxisEL = g.append("g")
+            .call(yAxis);
+        yAxisEL.attr("class", "axis axis--y")
             .append("text")
             .attr("class", "axis-title");
 
-        g.selectAll(".bar")
+        //declare bar variable to be used in resize function
+        let bar = g.selectAll(".bar")
             .data(data)
-            .enter().append("rect")
-            .attr("x", function (d) {
+            .enter().append("rect");
+
+        bar.attr("x", function (d) {
+            return x(d.index);
+        })
+        .attr("y", function (d) {
+            return y(d[field]);
+        })
+        .attr("width", x.bandwidth())
+        .attr("height", function (d) {
+            return height - y(d[field]);
+        })
+        .attr("fill", '#4682B4')
+        .on("mousemove", function(d){
+            tooltip_barchart
+                .style("left", d3.event.pageX - 50 + "px")
+                .style("top", d3.event.pageY - 70 + "px")
+                .style("display", "inline-block")
+                .html(d.index + '-> ' + d[field]);
+        })
+        .on("mouseout", function(d){ tooltip_barchart.style("display", "none");});
+
+        let gap = this.GAP;
+
+        //declare resize function
+        function resize() {
```

Comment:
I think it's a bit hacky and manual to do resize this way. In the function, there are basically almost all the code of the draw function (I know there is no draw function there but I mean the code doing the drawing). I think a few tips here:
- try to wrap the visualisation into a reusable piece of code like the old example from Mike Bostock (https://bost.ocks.org/mike/chart/) which you can call like calling the axis with width, height parameters.
- when you need to resize or basically redraw, just call the entire vis again (and if you do the update right, you can see the transition when resize rather than just removing elements and append them again).

I don't think I explain well but you can look at some the vises that I do here: the `pv-matrix` file is a reusable vis function. `matrix` just call the `pv-matrix` with data, width, height and other parameters. The 'drawing' and then 'resizing' code will just be `svg.datum(data).call(vis);` just like the same way we call how to draw an axis in d3 (`g.call(axis)`). 

Anyway, I don't think we should keep this any longer. Great work to spot which elements in the draw function to redraw. I hope it'll be easier for you to 'resize' the matrix (just need to reset the size for the vis object and call it again).